Bottom: d1e88aaada21530b076be7446f603ffd8a75ca70
Top:    8b3855324468a9cecdfa4a003bca1347ee1ce332
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-26 16:28:47 +0000

drc: HACK HACK HACK


---

diff --git a/src/find.c b/src/find.c
index d854b0b..8c737a3 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2610,7 +2610,12 @@ IsPolygonInPolygon (PolygonType *P1, PolygonType *P2)
 
               line.Point1.X = v->point[0];
               line.Point1.Y = v->point[1];
-              line.Thickness = 2 * Bloat;
+              line.Thickness = Bloat;
+              /* Another Bloat is added by IsLineInPolygon, making the correct
+               * 2x Bloat. Ideally we would change it there, but doing so
+               * breaks some other DRC checks which rely on the broken behaviour
+               * in IsLineInPolygon.
+               */
               line.Clearance = 0;
               line.Flags = NoFlags ();
               for (v = v->next; v != &c->head; v = v->next)
